File: siliconproperties/python_files/getDiffusionPotential.py
import numpy as np
from scipy import constants


def get_diffusion_potential(n_eff, temperature):
    ''' Diffusion potential [V] as a function of the effective doping
        concentration n_eff [10^12 / cm^3] and the temperature [K].

        Check/citetation of formulars needed!'''

    # [10^12/cm^3] empirical fit at K = 300 range
    N_i = 9.38 * 10 ** 7. * \
        (temperature / 300.)**2 * np.exp(-6884. / temperature)

    # Diffusion potential in thermal equilibrium [V]
    return constants.Boltzmann * temperature / constants.elementary_charge * np.log(n_eff**2. / N_i**2)


if __name__ == '__main__':
    import matplotlib.pylab as plt

    n_eff = np.linspace(0.1, 100., 1000.)

    # Plot diffusion potential
    for temperature in [200, 250, 300, 350]:
        plt.plot(n_eff, get_diffusion_potential(
            n_eff, temperature=temperature), linewidth=2.,
            label='T = %d' % temperature)
    plt.title(
        'Diffusion potential at thermal equilibrium in silicon')
    plt.xlabel('Effective doping concentration [$\mathrm{10^{12} / cm^3}}$]')
    plt.ylabel('Diffusion potential [$\mathrm{V}$]')
    plt.legend(loc=0)
    plt.grid()
    plt.savefig('DiffusionPotential.pdf', layout='tight')
    plt.show()

File: scarce/mathlab_files/_getDiffusionPotential.m
function Vbi = getDiffusionPotential(Neff, T)
    % Calculate the diffusion potential Vbi [V] as a function of the effective doping
    % concentration Neff [10^12 / cm^3] and te temperature [K]. 
    % Check/citetation of formulars needed!

    %constants
    Q_e = 1.602e-19;   %electron charge [C]
    K_b = 1.38e-23;   %Boltzmann Constant [J/K]
    
    N_i = 9.38.*10^7.*(T/300)^2.*exp(-6884/T);     % [10^12/cm^3] empirical fit at K = 300 range
    Vbi = K_b.*T./Q_e.*log(Neff.^2./N_i.^2);       % diffusion potential in thermal equilibrium [V]
 end %getTrappingTime  


